tests/utilities/test_enums.py

```diff
@@ -0,0 +1,10 @@
+from pytorch_lightning.utilities import DeviceType
+
+
+def test_consistency():
+    assert DeviceType.TPU not in ('GPU', 'CPU')
+    assert DeviceType.TPU in ('TPU', 'CPU')
+    assert DeviceType.TPU in ('tpu', 'CPU')
+    assert DeviceType.TPU not in {'GPU', 'CPU'}
+    assert DeviceType.TPU in {'TPU', 'CPU'}
```

Comment:
@tchaton The issue is that 'TPU' is hashed as uppercase but DeviceType.TPU is hashed as lower case, so I think we can just remove this line:
```suggestion
```

Reply:
but on the line it is exactly the same, right?

Reply:
One is `DeviceType.TPU in {'TPU'}`, which does:
- hash(DeviceType.TPU) -> has("TPU".lower()) == hash("TPU") -> False
The other is `DeviceType.TPU in {'tpu'}`:
- hash(DeviceType.TPU) -> has("TPU".lower()) == hash("tpu") -> True

So the issue is that you can't have a hash that is case invariant